Ada Examples/Examples Based On OpenGL Superbible/Simplest/src/initialization.adb

```diff
@@ -0,0 +1,123 @@
+
+with Interfaces.C;
+
+with Ada.Exceptions; use Ada.Exceptions;
+with Ada.Text_IO; use Ada.Text_IO;
+
+with gl.Context;
+with gl.Buffers;
+with GL.Errors;
+with GL.Objects.Vertex_Arrays;
+with GL.Objects.Shaders.Lists;
+with GL.Toggles;
+with GL.Types; use  GL.Types;
+
+with Glfw.Input.Mouse;
+with glfw.Windows;
+with glfw.Windows.Context;
+with glfw.Windows.Hints;
+
+with Shaders_Program;
+with Utilities;
+
+package body Initialization is
+
+    procedure Set_Window_Hints;
+    function Startup(Main_Window : Window_Types.tWindow;
+                     Rendering_Program : out GL.Objects.Programs.Program;
+                     Vertex_Array              : in out GL.Objects.Vertex_Arrays.Vertex_Array_Object) return Boolean;
+
+    --  -----------------------------------------------------------------------------------------------------------------------------
+
+    procedure Enable_Callbacks(thisWindow : in out Window_Types.tWindow) is
+    begin
+        thisWindow.Enable_Callback(glfw.Windows.Callbacks.Size);
+        thisWindow.Enable_Callback(glfw.Windows.Callbacks.Key);
+        thisWindow.Enable_Callback(glfw.Windows.Callbacks.Mouse_Button);
+        thisWindow.Enable_Callback(glfw.Windows.Callbacks.Mouse_Scroll);
+    end Enable_Callbacks;
+
+    --  -----------------------------------------------------------------------------------------------------------------------------
+
+   function Initialize(Main_Window : in out Window_Types.tWindow;
+                     Rendering_Program : out GL.Objects.Programs.Program;
+                       Vertex_Array            : in out GL.Objects.Vertex_Arrays.Vertex_Array_Object)
+                       return Boolean is
+
+        Window_Width       : Interfaces.C.Int := 800;
+        Window_Height      : Interfaces.C.Int := 600;
```

Comment:
Use `Glfw.Size` instead of `Interface.C.Int` here since that is the actual type taken by `Init`. Also consider making these variables constant.